backend/app/services/knowledge/permission_service.py

```
@@ -0,0 +1,363 @@
+    # 3. Personal KB - check owner or explicit permission
+    if kb.user_id == user_id:
+        return True
+
+    permission = (
+        db.query(Permission)
+        .filter(
+            Permission.kind_id == kb.id,
+            Permission.resource_type == "knowledge_base",
+            Permission.user_id == user_id,
+            Permission.permission_type.in_(["read", "download", "write", "manage"]),
+            Permission.is_active == True,
```

Ruff (0.14.14)

Avoid equality comparisons to `True`; use `Permission.is_active:` for truth checks

Replace with `Permission.is_active`

(E712)